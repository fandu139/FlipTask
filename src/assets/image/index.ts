/* eslint-disable global-require */
export const IMAGE_DEFAULT_EMPTY_LIST = require('./image_no_data.png');
