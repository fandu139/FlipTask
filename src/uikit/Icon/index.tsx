import React from 'react';
import { createIconSetFromIcoMoon } from 'react-native-vector-icons';
import IconConfig from '../../assets/icon/ColectionIcon.json';
import { TouchableOpacity, StyleProp, TextStyle } from 'react-native';
import AppStyles from '../../theme/appStyles';

interface Props {
  name?: string;
  size: number | undefined;
  style?: StyleProp<TextStyle>;
  color?: string | undefined;
  onPress?: () => void;
  disabled?: boolean;
  testID?: string;
  accessibilityLabel?: string;
}

const Icomoon = createIconSetFromIcoMoon(IconConfig);

const Icon: React.FC<Props> = ({
  name,
  size,
  style,
  color,
  onPress,
  disabled,
  testID,
  accessibilityLabel,
}: Props) => {
  if (name) {
    return (
      <TouchableOpacity
        disabled={disabled || onPress === undefined}
        onPress={onPress}
        testID={testID}
        accessibilityLabel={accessibilityLabel || testID}
        hitSlop={AppStyles.HitSlopValue}
      >
        <Icomoon
          name={name}
          size={size}
          style={{ ...(style as Record<string, unknown>), color }}
        />
      </TouchableOpacity>
    );
  }
  return null;
};

Icon.defaultProps = {
  name: undefined,
  color: undefined,
  style: {},
  disabled: false,
  onPress: undefined,
  testID: 'button-icon',
  accessibilityLabel: undefined,
};

export default Icon;
