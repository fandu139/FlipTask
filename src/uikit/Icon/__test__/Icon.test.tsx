import Icon from '../../Icon';
import React from 'react';
import { ICON_SHOW_PASSWORD } from '../../../assets/icon';
import { render } from '@testing-library/react-native';

describe('Icon component test', () => {
  describe('Icon snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<Icon name={ICON_SHOW_PASSWORD} size={16} color="black" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Icon without Name snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<Icon size={16} color="black" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
