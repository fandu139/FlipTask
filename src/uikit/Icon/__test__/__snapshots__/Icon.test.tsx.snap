// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component test Icon snapshot test should render correctly 1`] = `
<View
  accessibilityLabel="button-icon"
  accessible={true}
  focusable={false}
  hitSlop={
    Object {
      "bottom": 15,
      "left": 15,
      "right": 15,
      "top": 15,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="button-icon"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 16,
        },
        Object {
          "color": "black",
        },
        Object {
          "fontFamily": "ColectionIcon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î¤´
  </Text>
</View>
`;

exports[`Icon component test Icon without Name snapshot test should render correctly 1`] = `null`;
